### ISSUE TITLE
questions about the deformation degree

### ISSUE THREAD

**Issue by dandanli814** [TAMU-VITA/ShapeMatchingGAN#6]
Hi, I tested the pre-trained models  you provided, but I feel that the result like the water.gif effect is not very dynamic.  I want to make it more deformed.  
(1)What can I do? Do I need to change this "--scale -1 --scale_step 0.2 " or something else? 
(2) In your paper, it is mentioned that the range of parameter l is [0,1].  What is the reason for this?
Looking forward for your answer.
Thank you.

**Comment by williamyang1991** [comment]
The ``dynamic'' here does not means changing the deformation degree, it means that the water effects change within the same deformation degree. 
(1) To make it more deformed, just set ''--scale 1'', the maximam deformation degree in our model.
(2) I just linearly map the standard deviation of gaussian blur ranging from 8 to 24 into a normalized range of [0,1], which is easy to understand and adjust. You can map it to any ranges as you like. 
For more details, please refer to the paper.

**Comment by dandanli814** [comment]
Thank you very much. But I don't understand this " ”dynamic'' means that the water effects change within the same deformation degree".  
I think changing the degree of deformation and concat the corresponding generated images are to create a dynamic feeling(like .gif file). 
Maybe I am wrong. So how did you generate this file ./imgs/teaser-c.gif？

**Comment by williamyang1991** [comment]
Please see the Section 5.1 Network architecture.
''Since the structure map contains many saturated areas, we add gaussian noises onto the input of GS
and GT to avoid ambiguous problem. It also empowers our network to generate diversified results during testing.''

I just add random noise to the text images so that each time I will get a different results even I use the same deformation degree.
If you want to get a continously changing animation, you can just use two random noises, and interpolate them to get a set of noises to generate each frame of your animation.

**Comment by dandanli814** [comment]
So the reason why the animation effect can be generated is not because of changing the degree of deformation, but because of the addition of random noise?

**Comment by williamyang1991** [comment]
Yes. I think you might be confused with **dynamic typography** and **controllable text stylization**.

- The ''dynamic typography'' (teaser-d.gif, teaser-e.gif, teaser-f.gif) is an application of our method, by setting different noises but the same _l_.

- Our main contribution is the ''controllable text stylization'' (teaser-c.gif), by setting different _l_ but the same noise.

**Comment by dandanli814** [comment]
Yes, I was a bit confused before, and now I understand. Thank you very much. ^_^

**Comment by williamyang1991** [comment]
You are welcome.
Thank you for your interest in our work :)
